tools/dgeni/processors/categorizer.js

```diff
@@ -95,6 +99,67 @@ function filterDuplicateMembers(item, _index, array) {
   return array.filter((memberDoc, i) => memberDoc.name === item.name)[0] === item;
 }
 
+/** Sorts alphabetically by a member's name. */
+function sortByName(docA, docB) {
+  if (docA.name < docB.name) {
+    return -1;
+  }
+
+  if (docA.name > docB.name) {
+    return 1;
+  }
+
+  return 0;
+}
+
+/** Sort deprecated members to the end. */
+function sortByDeprecated(docA, docB) {
+  // Sort deprecated docs to the end
+  if (!docA.isDeprecated && docB.isDeprecated) {
+    return -1;
+  }
+
+  if (docA.isDeprecated && !docB.isDeprecated) {
+    return 1;
+  }
+
+  return 0;
+}
+
+/** Sorts methods by deprecated status and name. */
+function sortMethods(docA, docB) {
+  const deprecatedSort = sortByDeprecated(docA, docB);
+  if (!!deprecatedSort) {
+    return deprecatedSort
+  }
+
+  // Break ties by sorting alphabetically by name
+  return sortByName(docA, docB);
+}
+
+/** Sorts properties by deprecated status, decorator, and name. */
+function sortProperties(docA, docB) {
```

Comment:
Could you use the same function for properties and methods? All the checks for input/output should just return false for methods.